report.go

```diff
@@ -0,0 +1,106 @@
+package gomoss
+import (
+    "github.com/anaskhan96/soup"
+    "strings"
+    "os"
+    "regexp"
+    "fmt"
+)
+
+// Result of extracted data from Moss URLs
+type Result struct {
+    Codes []string
+    Percentages []string
+    Similarities []string
+}
+
+// FindMatches find all analysis related hrefs in HTML
+func FindMatches(baseURL string) []string {
+    resp, _ := soup.Get(baseURL)
+
+    doc := soup.HTMLParse(resp)
+    anchors := doc.FindAll("a")
+    exist := make(map[string]bool)
+    var urls []string
+    for _, anchor := range anchors {
+        url := anchor.Attrs()["href"]
+        if strings.Contains(url, "match") && !exist[url] {
+            exist[url] = true
+            urls = append(urls, url)
+        }
+    }
+    return urls
+}
+
+// DownloadURL into files
+func DownloadURL(url string, filename string) error {
+    resp, err := soup.Get(url)
+    if err != nil {
+        return err
+    }
+
+    f, err := os.Create(filename)
+    defer f.Close()
+    if err != nil {
+        return err
+    }
+
+    reg := regexp.MustCompile(`http:\/\/moss.stanford.edu\/results\/\d*\/`)
+    resp = reg.ReplaceAllString(resp, "")
+
+    f.Write([]byte(resp))
+
+    return nil
+}
+
+// DownloadMoss : Download all HTMLs from given moss URL recursively
```

Comment:
nit:

```go
// Download downloads all HTML files ...
```

in general you should always use `// Name verb-in-present-tense ...`